### PR TITLE
Update EnvironmentFiles and Unit Files to be Consistent

### PR DESCRIPTION
**Pull request by bensallen**
- Ensure each EnvironmentFile contains the appropriate *_OPTS variable
even if its empty by default. This serves as useful self-documentation
so users don't need to find the unit file on the system.
- alertmanager: Move --config.file and --storage.path args into
EnvironmentFile to make it configurable and consistent with the other
packages.
- prometheus: Move --config.file and --storage.local.path into
EnvironmentFile.
- prometheus2: Change OPTS value to be double quoted to being
consistent with other EnvironmentFiles.
- pushgateway: Bring PUSHGATEWAY_OPTS onto same line as ExecStart. No
need to split it as its under 80 char wide.
- sachet: SACHET_OPTS value needs to be quoted.
- sachet: Bring SACHET_OPTS onto the same line as ExecStart.

Signed-off-by: Benjamin S. Allen <bsallen@alcf.anl.gov>

### CODE REVIEW

**Comment by britcey** [comment]
I mostly like this but I believe this would be a breaking change for alertmanager if you already have a defaults file (as I do) that doesn't specify --config.file & --storage.path (same for prometheus v1).

Maybe better to keep those two in the unit file as well as adding them to the defaults, at least for a couple of releases.  This was how prometheus2 was configured (I think it's fine to remove those from that unit file).

We'd also have to bump the release value on each package.

**Comment by bensallen** [comment]
Appears at least for AlertManager, it doesn't allow repeated arguments:

```
# /usr/bin/alertmanager --config.file=/etc/prometheus/alertmanager.yml --storage.path=/var/lib/prometheus/alertmanager --config.file=/etc/prometheus/alertmanager.yml.asdf
alertmanager: error: flag 'config.file' cannot be repeated, try --help
```

So keeping the arguments in the Unit files and specifying them in defaults won't work.

We could set two dedicated Environment variables via the unit file, eg.

```
[Service]
	Environment=ALERTMANAGER_CONFIG="--config.file=/etc/prometheus/alertmanager.yml"
	Environment=ALERTMANAGER_STORAGE="--storage.path=/var/lib/prometheus/alertmanager"
	EnvironmentFile=-/etc/default/alertmanager
	ExecStart=/usr/bin/alertmanager $ALERTMANAGER_CONFIG $ALERTMANAGER_STORAGE $ALERTMANAGER_OPTS
...
```

This will let users override the settings still via the defaults file.

**Comment by britcey** [comment]
I think as a way not to break existing installations, that seems reasonable - we can make this one package non-standard in that way going forward or just advertise that it will, in a release or two, require those two flags to be part of ALERTMANAGER_OPTS

**Comment by bensallen** [comment]
@britcey Ok updated with this approach for alertmanager and prometheus v1. Let me know if you'd like me to bump versions.

**Comment by zoonage** [comment]
Looks good to me!

**Comment by britcey** [comment]
my $0.02 is this is OK w/o a package version bump.

**Comment by mrichar1** [comment]
Just hit the storage.path issue, and came here to suggest this change. Is there any chance of this PR being tidied up and committed?

File: alertmanager/alertmanager.service [hunk 1]
```diff
@@ -6,12 +6,11 @@ Documentation=https://github.com/prometheus/alertmanager
 After=network.target
 
 [Service]
+Environment=ALERTMANAGER_CONFIG="--config.file=/etc/prometheus/alertmanager.yml"
+Environment=ALERTMANAGER_STORAGE="--storage.path=/var/lib/prometheus/alertmanager"
```
Comment (zoonage) [inline alertmanager/alertmanager.service:10]: Should we just collapse these into a single `ALERTMANAGER_OPTS`?

**Pull request closed without merging**